### PR TITLE
[ADDS] Dusk Tests

### PR DESCRIPTION
**Pull request by MWahhab**

### CODE REVIEW

File: tests/Browser/ExampleTest.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Tests\Browser;
+
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Laravel\Dusk\Browser;
+use Tests\DuskTestCase;
+
+class ExampleTest extends DuskTestCase
+{
```
Comment (lolmeherti) [inline tests/Browser/ExampleTest.php]: do not commit this, wouldn't work anyway.

File: tests/Browser/BoardInteractionTest.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+               ->assertSeeIn('span.sr-only', 'Delete')
+               ->click('button.text-red-500')
+
+               ->assertSee("Confirm Deletion")
+               ->assertSee("Are you sure you want to delete this board? This action cannot be undone.")
+
+               ->press("Delete")
+
+               ->assertDontSee("Auto Gen Test Board");
+
```
Comment (lolmeherti) [inline tests/Browser/BoardInteractionTest.php]: missing logout

File: tests/Browser/LoginTest.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        }
+
+        throw new \Exception('Something has gone wrong. The test user already exists!');
+    }
+
+    private function deleteLoginTestUser(): void
+    {
+        User::where('email', 'test1986968215513@test.com')->delete();
+    }
+}
```
Comment (lolmeherti) [inline tests/Browser/LoginTest.php]: not testing logout

File: tests/Browser/Pages/HomePage.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Tests\Browser\Pages;
```
Comment (lolmeherti) [inline tests/Browser/Pages/HomePage.php]: why do we need this?

File: tests/Browser/BoardInteractionTest.php [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+class BoardInteractionTest extends DuskTestCase
+{
+    /**
+     * Test a successful board creation
+     */
+    public function testSuccessfulCreation(): void
+    {
+        $this->browse(function (Browser $browser) {
+            $this->reachDashboard("login");
+
```
Comment (lolmeherti) [inline tests/Browser/BoardInteractionTest.php]: in the tests below you're testing if the reachDashboard("login") returns a valid user id, here you didn't do that.

File: tests/Browser/RegistrationTest.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Tests\Browser;
+
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Laravel\Dusk\Browser;
+use Tests\DuskTestCase;
+
+class RegistrationTest extends DuskTestCase
+{
```
Comment (lolmeherti) [inline tests/Browser/RegistrationTest.php]: what is the meaning of this?

File: tests/Browser/RegistrationTest.php~ [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Tests\Browser;
+
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Laravel\Dusk\Browser;
+use Tests\DuskTestCase;
+
+class RegistrationTest extends DuskTestCase
+{
+    /**
```
Comment (lolmeherti) [inline tests/Browser/RegistrationTest.php~]: why is this file duplicated?

**Review by lolmeherti — CHANGES_REQUESTED** [review]
please have a look at the comments and clarify/change where needed

File: tests/Browser/BoardInteractionTest.php (hunk 2 above)
Comment (MWahhab) [reply tests/Browser/BoardInteractionTest.php]: added logout

File: tests/Browser/RegistrationTest.php (hunk 6 above)
Comment (MWahhab) [reply tests/Browser/RegistrationTest.php]: issue with ide when committing, fixed

File: tests/Browser/RegistrationTest.php~ (hunk 7 above)
Comment (MWahhab) [reply tests/Browser/RegistrationTest.php~]: issue with ide, fixed

File: tests/Browser/BoardInteractionTest.php (hunk 5 above)
Comment (MWahhab) [reply tests/Browser/BoardInteractionTest.php]: resolved

File: tests/Browser/LoginTest.php (hunk 3 above)
Comment (MWahhab) [reply tests/Browser/LoginTest.php]: incorporated it into the login test and took screenshots

File: tests/Browser/Pages/HomePage.php (hunk 4 above)
Comment (MWahhab) [reply tests/Browser/Pages/HomePage.php]: removed

**Pull request closed without merging**